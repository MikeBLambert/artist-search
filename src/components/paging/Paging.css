#pagingBar {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  font-family: 'Black Ops One', cursive;
  color: black;

  button {
    height: 7vh;
    margin: 3em;
    width: 10vw;
    background: lightgrey;
    border: 2px solid black;
    border-radius: 15px;
    box-shadow: 1px 1px grey;

  }
}