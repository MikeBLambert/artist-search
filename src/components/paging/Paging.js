import React, { Fragment } from 'react';
import PropTypes from 'prop-types';


const Paging = ({ currentPage, totalPages, updatePage }) => {

  return (
    <Fragment>
      {currentPage > 1 && <button onClick={()=>updatePage(currentPage - 1)}>Previous</button>}
      <h1>{currentPage} out of {totalPages}</h1>
      {totalPages > currentPage && <button onClick={()=>updatePage(currentPage + 1)}>Next</button>}
    </Fragment>
  );
};

// Paging.propTypes = {
//   currentPage: PropTypes.number.isRequired,
//   totalPages: PropTypes.number.isRequired,
//   updatePage: PropTypes.func.isRequired
// };


export default Paging;
