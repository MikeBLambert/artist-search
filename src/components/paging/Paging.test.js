import React from 'react';
import Paging from './Paging';
import { shallow } from 'enzyme';

describe('Paging component', () => {


  it('renders a Paging component', () => {
    const currentPage = 3;
    const totalPages = 10;
    const updatePage = jest.fn();
    const wrapper = shallow(<Paging currentPage={currentPage} totalPages={totalPages} updatePage={updatePage} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('has the current page rendered', () => {
    const currentPage = 3;
    const totalPages = 10;
    const updatePage = jest.fn();
    const wrapper = shallow(<Paging currentPage={currentPage} totalPages={totalPages} updatePage={updatePage} />);
    expect(wrapper.html()).toContain(currentPage);
  });
  it('displays total pages', () => {
    const totalPages = 10;
    const updatePage = jest.fn();
    const wrapper = shallow(<Paging currentPage={1} totalPages={totalPages} updatePage={updatePage} />);
    expect(wrapper.html()).toContain(totalPages);
  });

  it('has a previous button when current page is greater than 0', () => {
    const updatePage = jest.fn();
    const wrapper = shallow(<Paging currentPage={2} totalPages={10} updatePage={updatePage} />);
    expect(wrapper.find('button').at(0).text()).toEqual('Previous');
  });

  it('has a next button', () => {
    const updatePage = jest.fn();
    const wrapper = shallow(<Paging currentPage={2} totalPages={10} updatePage={updatePage} />);
    expect(wrapper.find('button').at(1).text()).toEqual('Next');
  });

  it('decrements currentPage when previous button is clicked', () => {
    const updatePageFn = jest.fn();
    const wrapper = shallow(<Paging currentPage={10} totalPages={10} updatePage={updatePageFn}/>);
    wrapper.find('button').at(0).simulate('click');
    expect(updatePageFn.mock.calls[0][0]).toEqual(9);
  });

  it('increments currentPage when next button is clicked', () => {
    const updatePageFn = jest.fn();
    const wrapper = shallow(<Paging currentPage={10} totalPages={20} updatePage={updatePageFn}/>);
    wrapper.find('button').at(1).simulate('click');
    expect(updatePageFn.mock.calls[0][0]).toEqual(11);
  });

  it('doesn\'t display previous button when currentPage is 1', () => {
    const updatePage = jest.fn();
    const wrapper = shallow(<Paging currentPage={1} totalPages={20} updatePage={updatePage} />);
    expect(wrapper.find('button')).toHaveLength(1);
    expect(wrapper.find('button').at(0).text()).toEqual('Next');
  });

  it('doesn\'t display next button when currentPage === nextPage', () => {
    const updatePage = jest.fn();
    const wrapper = shallow(<Paging currentPage={20} totalPages={20} updatePage={updatePage} />);
    expect(wrapper.find('button')).toHaveLength(1);
    expect(wrapper.find('button').at(0).text()).toEqual('Previous');
  });
});
