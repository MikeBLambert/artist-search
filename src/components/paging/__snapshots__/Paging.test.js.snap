// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging component renders a Paging component 1`] = `
<Fragment>
  <button
    onClick={[Function]}
  >
    Previous
  </button>
  <h1>
    3
     out of 
    10
  </h1>
  <button
    onClick={[Function]}
  >
    Next
  </button>
</Fragment>
`;
